src/Tasks/Hash.cs

```diff
@@ -52,7 +62,12 @@ public override bool Execute()
 
                     using (var stringBuilder = new ReuseableStringBuilder(Math.Max(concatenatedItemStringSize, hashStringSize)))
                     {
-                        foreach (var item in ItemsToHash)
+                        IEnumerable<ITaskItem> items = ItemsToHash;
+                        if (IgnoreOrder)
+                        {
+                            items = items.OrderBy(o => o.ItemSpec, IgnoreCase ? StringComparer.OrdinalIgnoreCase : StringComparer.Ordinal);
```

Comment:
Can we use .Sort here instead of .OrderBy?

Reply:
While I'm not sure there'd be an observable difference sorting the input `ITaskItem[]` in place, the explicit new enuerable created by `OrderBy` feels like a better approach to me.

Reply:
nit: No need to use `IgnoreCase` here, it's only making the sorting operation slower. The result will be correct even if items are always sorted case sensitive.

Reply:
I actually think that in place sort should be better here than Linq. Hash does show up in profiles, and we should really care about allocations. From my reading of binlogs Hash is used quite a lot to hash large arrays. And OrderBy is just insanely wasteful:
https://github.com/KirillOsenkov/Benchmarks/blob/516b5bde9bd7bdc59a9fdd5b6436b26480d7d795/src/Tests/SortDictionary.cs#L10-L17